docs-src/app-dev-context/mutable-side-effects.md

```diff
@@ -0,0 +1,23 @@
+---
+id: mutable-side-effects
+title: Mutable Side Effects
+description: Mutable Side Effects are a method of execution to produce nondeterministic code.
+sidebar_label: Mutable Side Effects
+tags:
+  - guide-context
+---
+
+Mutable Side Effects execute the provided function once, and then it looks up the History of the value with the given Workflow ID.
```

Comment:
Effects multiple, but function singular? Doesn't make sense to me. Probably better to rephrase in singulars.

Reply:
"then it looks up the History of the value with the given Workflow ID."
I don't understand what it means, it doesn't sound right. It doesn't look up a history of the value. Workflow ID is irrelevant here.

This segment from the javadoc describes it better:

"Executes the provided function once, records its result into the workflow history. The recorded
   * result on history will be returned without executing the provided function during replay. This
   * guarantees the deterministic requirement for workflow as the exact same result will be returned
   * in replay."

Reply:
> Mutable Side Effects execute the provided function once, and then it looks up the History of the value with the given Workflow ID.

Should be just ID or Side Effect ID, Workflow ID is incorrect.

"execute the provided function once"

This is unclear. Should be just "executes the provided function during original execution (not replay)"